graph/path/yen_ksp_test.go

```diff
@@ -0,0 +1,113 @@
+// Copyright ©2018 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package path
+
+import (
+	"math"
+	"reflect"
+	"testing"
+
+	"gonum.org/v1/gonum/graph"
+	"gonum.org/v1/gonum/graph/simple"
+)
+
+var yenShortestPathTests = []struct {
+	name  string
+	graph func() graph.WeightedEdgeAdder
+	edges []simple.WeightedEdge
+
+	query     simple.Edge
+	k         int
+	wantPaths [][]int64
+}{
+	// Positive weighted graphs.
+	// https://en.wikipedia.org/wiki/Yen%27s_algorithm
```

Comment:
This link should be on the element for the wikipedia example, and be `https://en.wikipedia.org/w/index.php?title=Yen%27s_algorithm&oldid=841018784#Example`.